src/Perpetuum/Zones/LineOfSight.cs

```diff
@@ -98,8 +98,20 @@ public static LOSResult IsInLineOfSight(this IZone zone, Position start, Unit ta
         }
 
         [NotNull]
-        private static LOSResult IsInLineOfSight(IZone zone,Vector3 origin,Vector3 direction,float distance,bool ballistic)
+        private static LOSResult IsInLineOfSight(IZone zone, Vector3 origin, Vector3 direction, float distance, bool ballistic)
         {
+            if (distance == 0.0)
```

Comment:
You're going to need an "Approximately Equal" function. Due to floating point error this might not be true, but is instead virtually zero i.e. 0.000000000000000000123

Something like this:
```
    public static bool IsApproximatelyEqual(this float left, float right, float precision = 1e-7f)
    {
      var result = Math.Abs(left - right);
      return result <= precision;
    }
```

You would then call it like `if (distance.IsApproximatelyEqual(0, 0.1))` to check if the value is 0 to within a margin of error of 0.1. i.e. 0.05 would be true, 0.09 would be true, 0.1 **would also be true**, 0.10001 would be false

Reply:
Yes I was considering an equal within epsilon, but I think the absolute boundary of this case is when distance (or magnitude of direction vector) is 0, prior vector math makes its components NaN and thats where things go wrong.
As soon as there is a distance, this is a direction, and seems to be ok.

But I agree, it feels safer to just get a bit of a margin, in case other operations still turn out 0->NaN'ing vectors, but distance isn't exactly 0.